src/components/Map/EditableMapV2.tsx

```diff
@@ -192,14 +192,18 @@ export default function EditableMap({
           return;
         }
 
-        initializePopupInfo({
-          id,
-          lng,
-          lat,
-          properties,
-          sourceId,
-          active: true,
-        });
+        // the timeouts are to avoid a race condition upon click vs double-click both of which have
+        // popup triggers but with a priority (this popup gets overridden by the one from zone creation)
+        setTimeout(() => {
```

Comment:
Is there another way to do this? It seems a bit hacky and I suspect that it will make future (hopefully) automated tests fail intermittently.

Reply:
Yes, this is sad. We have two click events that do different things - differentiating between the two is getting tricky (click on zone in map vs click to finish new polygon/zone on map) - both popup the info box on the clicked zone. The set timeouts allowed for clicking across zones to popup the info box for that zone, without having to close the previous info box. I can restrict the UX to not popup a new info box as long as one is already open - user will have to close the current info box before opening the new one. Not a big sacrifice.

Reply:
I don't want to hold you up or hurt the UX (I think making them close the box before they can open another one sounds less ideal than the current setup), sounds like a necessary evil at the moment.

Reply:
Gist of the problem:
1. the mapbox polygon draw control has click events that trigger creation of polygon and popping up of the info box
2. the same click is also captured by the map and calls onClose on the popup info box (equivalent to clicking outside the box to close it).. hence all that earlier hacky logic to workaround this
I can't find a way to get a hold of the click event from step 1. --> original plan was to stopPropagation of that event (since it is only relevant to polygon creation).. but there's no hook to this event, sigh. That would have been ideal.

Reply:
Created [jira](https://terraformation.atlassian.net/browse/SW-4760) to follow up again later.